Mathlib/RingTheory/FractionalIdeal/LocalizedAtPrime.lean

```diff
@@ -0,0 +1,210 @@
+/-
+Copyright (c) 2024 James Sundstrom. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: James Sundstrom
+-/
+import Mathlib.RingTheory.FractionalIdeal.Extended
+import Mathlib.RingTheory.Localization.AtPrime
+import Mathlib.RingTheory.DedekindDomain.Ideal
+import Mathlib.Algebra.Module.LocalizedModule
+
+/-!
+# Localization of a fractional ideal at a prime
+
+This file defines the localization of a fractional ideal at a prime ideal.
+
+## Main definition
+
+* `FractionalIdeal.localizationAtPrime`: The localization of a fractional ideal at a prime ideal.
+
+## Main results
+
+* `coe_localizedAtPrime`: The localization of `I` is `Submodule.span Aₚ I`.
+* `localizedAtPrime_add`: Localizing commutes with addition.
+* `localizedAtPrime_mul`: Localizing commutes with multiplication.
+* `localizedAtPrime_div`: Localizing commutes with division for finitely-generated denominator.
+* `localizedAtPrime_inv`: Localizing commutes with inverses for finitely-generated `I`.
+* `not_le_of_localizedAtPrime_eq_one`: If `I.localizedAtPrime P = 1`, then `¬ I ≤ P`.
+
+## Tags
+
+fractional ideal, fractional ideals, localization
+-/
+
+open IsLocalization Localization Algebra FractionalIdeal Submodule nonZeroDivisors Finset Module
+
+variable {A : Type*} [CommRing A] [IsDomain A] (P : Ideal A) [P.IsPrime]
+variable (I : FractionalIdeal A⁰ (FractionRing A)) (J : FractionalIdeal A⁰ (FractionRing A))
+
+local notation "Aₚ" => Localization.AtPrime P
+local notation "K" => FractionRing A
+local notation "hf" => nonZeroDivisors_le_comap_nonZeroDivisors_of_injective _ <|
+  IsLocalization.injective Aₚ P.primeCompl_le_nonZeroDivisors
```

Comment:
I wanted to define it in greater generality, but I couldn't find an easy way to get it to work. If I recall correctly, the problem was that `K` needs to be an `Aₚ`-algebra, which can't (currently) be inferred when they're defined using `IsFractionRing` and `IsLocalization`.

Reply:
To answer the other question, the only follow-up work I did with this was the proof of `isDedekindDomain ↔ isDedekindDomainDvr` linked above. In that specific case, being stuck with `FractionRing` rather than `IsFractionRing` didn't cause any problems.

Reply:
> I wanted to define it in greater generality, but I couldn't find an easy way to get it to work. If I recall correctly, the problem was that `K` needs to be an `Aₚ`-algebra, which can't (currently) be inferred when they're defined using `IsFractionRing` and `IsLocalization`.

Is the issue that we are missing some instances? Can you switch to the `Is` point of view and add them, or give some idea about problems which you can't solve using the `Is` approach?